src/Controller/Admin/Improve/Design/LinkWidgetController.php

```diff
@@ -0,0 +1,252 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\LinkList\Controller\Admin\Improve\Design;
+
+use PrestaShop\Module\LinkList\Core\Grid\LinkBlockGridFactory;
+use PrestaShop\Module\LinkList\Core\Search\Filters\LinkBlockFilters;
+use PrestaShop\Module\LinkList\Form\LinkBlockFormDataProvider;
+use PrestaShop\Module\LinkList\Repository\LinkBlockRepository;
+use PrestaShop\PrestaShop\Adapter\Entity\PrestaShopDatabaseException;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Grid\Presenter\GridPresenter;
+use PrestaShopBundle\Controller\Admin\AbstractModuleFrameworkBundleAdminController;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\ModuleActivated;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class LinkWidgetController
+ * @package PrestaShop\Module\LinkList\Controller\Admin\Improve\Design
+ *
+ * @ModuleActivated(moduleName="ps_linklist", redirectRoute="admin_module_manage")
+ */
+class LinkWidgetController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @Template("@Modules/ps_linklist/views/templates/admin/link_block/list.html.twig")
+     *
+     * @return array
+     */
+    public function listAction(Request $request)
+    {
+        //Get hook list, then loop through hooks setting it in in the filter
+        /** @var LinkBlockRepository $repository */
+        $repository = $this->get('prestashop.module.link_block.repository');
+        $hooks = $repository->getHooksWithLinks();
+
+        $filtersParams = $this->buildFiltersParamsByRequest($request);
+
+        /** @var LinkBlockGridFactory $linkBlockGridFactory */
+        $linkBlockGridFactory = $this->get('prestashop.module.link_block.grid.factory');
+        $grids = $linkBlockGridFactory->getGrids($hooks, $filtersParams);
+
+        /** @var GridPresenter $gridPresenter */
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
```

Comment:
The presenter could contains a presentCollection method?

Reply:
I am not sure if it is useful This behaviour is really specific to this module because we have multiple grid per hooks In most cases there is only one grid, even my `LinkBlockGridFactory` is kinda out of the regular scope
But if we think that we will regularly have multiple grids in a single page then this might be useful indeed, @mickaelandrieu wdyt?